book/src/dev/rfcs/0010-v5-transaction.md

```diff
@@ -0,0 +1,247 @@
+- Feature Name: `v5_transaction`
+- Start Date: 2021-03-11
+- Design PR: [ZcashFoundation/zebra#1886](https://github.com/ZcashFoundation/zebra/pull/1886)
+- Zebra Issue: [ZcashFoundation/zebra#1863](https://github.com/ZcashFoundation/zebra/issues/1863)
+
+# Summary
+[summary]: #summary
+
+Network Upgrade number 5 (`NU5`) introduces a new transaction type (transaction version 5). This document is a proposed design for implementing such a transaction version.
+
+# Motivation
+[motivation]: #motivation
+
+The Zebra software wants to be a protocol compatible Zcash implementation. One of the tasks to do this includes the support of the new version 5 transactions that will be implemented in Network Upgrade 5 (NU5).
+
+# Definitions
+[definitions]: #definitions
+
+- `NU5` - the 5th Zcash network upgrade, counting from the `Overwinter` upgrade as upgrade zero.
+- `Orchard` - a new shielded pool introduced in `NU5`.
+- `Sapling` - a new shielded pool introduced in the 1st network upgrade. (`Sapling` is also the name of that network upgrade, but this RFC is focused on the `Sapling` shielded pool.)
+- `orchard data` - Data types needed to support orchard transactions.
+- `sapling data` - Data types needed to support sapling transactions.
+- `orchard transaction version` - Transactions that support orchard data. Currently only V5.
+- `sapling transaction version` - Transactions that support sapling data. Currently V4 and V5 but the data is implemented differently in them.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+V5 transactions are described by the protocol in the second table of [Transaction Encoding and Consensus](https://zips.z.cash/protocol/nu5.pdf#txnencodingandconsensus).
+
+We need the raw representation to Serialize/Deserialize the transaction however Zebra uses its own representation internally.
+
+V5 transactions are the only ones that will support orchard transactions with `Orchard` data types.
+
+V4 and V5 transactions both support sapling, but the underying data structures are different. So need to make the sapling data types generic over the V4 and V5 structures.
+
+The order of some of the fields changed from V4 to V5. For example the `lock_time` and `expiry_height` were moved above the transparent inputs and outputs.
+
+All of the changes proposed in this document are only to the `zebra-chain` crate.
+
+To highlight changes most of the document comments from the code snippets in the [reference section](#reference-level-explanation) were removed.
```

Comment:
Non-blocking:

If we're all happy with the reference-level design, we should summarise the main parts of that design in the guide-level section.

Reply:
Proposed structure:
* Sapling Changes
* Orchard Additions
* Other V5 Changes

Each subsection should have 1-3 sentences.

Reply:
```suggestion
# Guide-level explanation
[guide-level-explanation]: #guide-level-explanation

V5 transactions are described by the protocol in the second table of [Transaction Encoding and Consensus](https://zips.z.cash/protocol/nu5.pdf#txnencodingandconsensus).

All of the changes proposed in this document are only to the `zebra-chain` crate.

To highlight changes most of the document comments from the code snippets in the [reference section](#reference-level-explanation) were removed.

## Sapling Changes Overview

V4 and V5 transactions both support sapling, but the underlying data structures are different. So need to make the sapling data types generic over the V4 and V5 structures.

In V4, anchors are per-spend, but in V5, they are per-transaction.

For consistency, also we move some fields into the `ShieldedData` type, and rename some fields and types.

## Orchard Additions Overview

V5 transactions are the only ones that will support orchard transactions with `Orchard` data types.

Orchard uses `Halo2Proof`s with corresponding signature type changes. Each Orchard `Action` contains a spend and an output. Placeholder values are substituted for unused spends and outputs.

## Other Transaction v5 Changes

The order of some of the fields changed from V4 to V5. For example the `lock_time` and `expiry_height` were moved above the transparent inputs and outputs.
```